hack/dockerfiles/generated-files.Dockerfile

```diff
@@ -1,48 +1,50 @@
 # syntax=docker/dockerfile-upstream:master
 
-# protoc is dynamically linked to glibc to can't use golang:1.10-alpine
-FROM golang:1.19-buster AS gobuild-base
-
-RUN apt-get update && apt-get --no-install-recommends install -y \
-	unzip \
-	&& true
-
-# https://github.com/golang/protobuf/blob/v1.3.5/.travis.yml#L15
-ARG PROTOC_VERSION=3.11.4
-ARG TARGETOS TARGETARCH
-RUN set -e; \
-	arch=$(echo $TARGETARCH | sed -e s/amd64/x86_64/ -e s/arm64/aarch_64/); \
-	wget -q https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-${TARGETOS}-${arch}.zip && unzip protoc-${PROTOC_VERSION}-${TARGETOS}-${arch}.zip -d /usr/local
-
-ARG GOGO_VERSION=v1.3.2
-RUN --mount=target=/root/.cache,type=cache GO111MODULE=on go install \
-	github.com/gogo/protobuf/protoc-gen-gogo@${GOGO_VERSION} \
-	github.com/gogo/protobuf/protoc-gen-gogofaster@${GOGO_VERSION} \
-	github.com/gogo/protobuf/protoc-gen-gogoslick@${GOGO_VERSION}
-
-ARG PROTOBUF_VERSION=v1.3.5
-RUN --mount=target=/root/.cache,type=cache GO111MODULE=on go install \
-	github.com/golang/protobuf/protoc-gen-go@${PROTOBUF_VERSION}
-
+ARG GO_VERSION="1.19"
+ARG PROTOC_VERSION="3.11.4"
+
+# protoc is dynamically linked to glibc so can't use alpine base
+FROM golang:${GO_VERSION}-buster AS base
+RUN apt-get update && apt-get --no-install-recommends install -y git unzip
+ARG PROTOC_VERSION
+ARG TARGETOS
+ARG TARGETARCH
+RUN <<EOT
+  set -e
+  arch=$(echo $TARGETARCH | sed -e s/amd64/x86_64/ -e s/arm64/aarch_64/)
+  wget -q https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-${TARGETOS}-${arch}.zip
+  unzip protoc-${PROTOC_VERSION}-${TARGETOS}-${arch}.zip -d /usr/local
+EOT
 WORKDIR /go/src/github.com/moby/buildkit
 
-# Generate into a subdirectory because if it is in the root then the
-# extraction with `docker export` ends up putting `.dockerenv`, `dev`,
-# `sys` and `proc` into the source directory. With this we can use
-# `tar --strip-components=1 generated-files` on the output of `docker
-# export`.
-FROM gobuild-base AS generated
-RUN mkdir /generated-files
-RUN --mount=target=/tmp/src \
-	cp -r /tmp/src/. . && \
-	git add -A && \
-	go generate -mod=vendor -v ./... && \
-	git ls-files -m --others -- **/*.pb.go | tar -cf - --files-from - | tar -C /generated-files -xf -
+FROM base AS tools
+RUN --mount=type=bind,target=.,rw \
+    --mount=type=cache,target=/root/.cache \
+    go install $(sed -n -e 's|^\s*_\s*"\(.*\)".*$|\1| p' ./util/pb_tools.go)
```

Comment:
I don't really like the `./util/pb_tools.go` file - am I right to assume that that file would need to exist to be able to use this `Dockerfile` in a scenario like https://github.com/docker/buildx/pull/1296#discussion_r1004506672?

Can we just install the packages as before, but without version numbers to load them from `go.mod`?

Reply:
Not just for the Dockerfile but to vendor these tools otherwise they cannot be inferred from `go.mod`.

Reply:
This will be painful for anyone importing buildkit. I've seen the same mess before in https://github.com/google/certificate-transparency-go/blob/v1.1.0/tools.go making it so that if you imported from that library it pulled in the whole world. They have removed it now in the last version.

Reply:
Moved to a subpackage so it doesn't import extra deps. See also https://github.com/google/trillian/issues/2196